src/bbsecondary.c

```diff
@@ -158,6 +158,12 @@ bool start_secondary(bool need_secondary) {
       "-modulepath", bb_config.mod_path, // keep last
       NULL
     };
+    char **argvp;
+    bb_log(LOG_DEBUG, "X server command line:");
+    for (argvp = x_argv; argvp; argvp++) {
```

Comment:
This causes a SEGFAULT. The condition should be "argvp && *argvp" to hit the NULL character guard at the end of x_argv, otherwise the pointer keeps getting incremented until it eventually causes a protection fault.